kroxylicious/src/main/java/io/kroxylicious/proxy/internal/DefaultFilterContext.java

```diff
@@ -165,21 +170,28 @@ public <T extends ApiMessage> CompletionStage<T> sendRequest(short apiVersion, A
     /**
      * Forward a request to the next filter in the chain
      * (or to the downstream client).
+     *
+     * @param header The header
      * @param response The message
      */
     @Override
-    public void forwardResponse(ApiMessage response) {
+    public void forwardResponse(ResponseHeaderData header, ApiMessage response) {
         // check it's a response
         String name = response.getClass().getName();
         if (!name.endsWith("ResponseData")) {
             throw new AssertionError("Attempt to use forwardResponse with a non-response: " + name);
         }
         if (decodedFrame instanceof RequestFrame) {
-            forwardShortCircuitResponse(response);
+            forwardShortCircuitResponse(header, response);
         }
         else {
             // TODO check we've not forwarded it already
-
+            if (decodedFrame.body() != response) {
```

Comment:
I added these checks to match the `forwardRequest` case, this is because we currently only support mutating the request/response passed to the filter (except in the short-circuit case where we forwardResponse before it's sent to the broker)

Reply:
Created #303 to add support for new instances